central/productusage/datastore/securedunits/datastore_impl.go

```diff
@@ -17,26 +20,44 @@ var (
 )
 
 type dataStoreImpl struct {
+	store     store.Store
 	clusterDS clusterDataStore
 	cache     cache.Cache
 }
 
 var _ DataStore = (*dataStoreImpl)(nil)
 
-// Get returns the object, if it exists from the store.
-func (ds *dataStoreImpl) Walk(ctx context.Context, _ *types.Timestamp, _ *types.Timestamp, _ func(m *storage.SecuredUnits) error) error {
+// Walk returns the object, if it exists from the store.
+func (ds *dataStoreImpl) Walk(ctx context.Context, from *types.Timestamp, to *types.Timestamp, fn func(*storage.SecuredUnits) error) error {
 	if err := sac.VerifyAuthzOK(usageSAC.ReadAllowed(ctx)); err != nil {
-		return errors.Wrap(err, "cannot permit to get usage data")
+		return errors.Wrap(err, "cannot permit to walk through usage data")
```

Comment:
A rule of thumb for messages in Wrap and CausedBy calls: what's the extra information _not available to the caller_ that might be useful in the context of this error? Here the caller knows what the call Walk on "usage data" and permission problem is clear from `err`.

Reply:
What's new in "cannot permit" that is not in `access to resource denied`?

Reply:
not much